From 59d9df3007e6cff8de4e576f4c415d0deaea8f16 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Mon, 27 Jun 2016 06:37:44 +0000
Subject: [PATCH] serial_core: fix wrong type cast

Fix wrong type cast or it will occur the build warning as below:

kernel-source/drivers/tty/serial/serial_core.c:749:26: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   new_info->iomem_base = (unsigned char *)uport->mapbase;
                          ^

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/tty/serial/serial_core.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/tty/serial/serial_core.c b/drivers/tty/serial/serial_core.c
index 7b9ec1b..d293536 100644
--- a/drivers/tty/serial/serial_core.c
+++ b/drivers/tty/serial/serial_core.c
@@ -746,7 +746,7 @@ static int uart_set_info(struct tty_struct *tty, struct tty_port *port,
 	 * IO port changes.
 	 */
 	if ((unsigned long)new_info->iomem_base == 0xffffffff)
-		new_info->iomem_base = (unsigned char *)uport->mapbase;
+		new_info->iomem_base = (void *)(unsigned long)uport->mapbase;
 
 	change_port = !(uport->flags & UPF_FIXED_PORT)
 		&& (new_port != uport->iobase ||
-- 
1.7.5.4

